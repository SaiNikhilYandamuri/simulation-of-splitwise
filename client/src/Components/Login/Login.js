import axios from 'axios';
import { Component, React } from 'react';
import alert from 'alert';
import '../Landing/Landing.css';

class Login extends Component {
  constructor(props) {
    super(props);

    this.state = {
      email: '',
      password: '',
    };
    this.emailChangeHandler = this.emailChangeHandler.bind(this);
    this.passwordChangeHandler = this.passwordChangeHandler.bind(this);
    this.submitLogin = this.submitLogin.bind(this);
    this.verifyEmailAddressFormat = this.verifyEmailAddressFormat.bind(this);
  }

  emailChangeHandler = (e) => {
    this.setState({
      email: e.target.value,
    });
  };

  passwordChangeHandler = (e) => {
    this.setState({
      password: e.target.value,
    });
  };

  verifyEmailAddressFormat = (email) => {
    if (email.includes('@') && email.includes('.com')) {
      return true;
    }
    return false;
  };

  submitLogin(e) {
    e.preventDefault();
    const { email, password } = this.state;
    const data = {
      email,
      password,
    };

    if (this.verifyEmailAddressFormat(data.email)) {
      axios
        .post('http://localhost:4000/login', data)
        .then((response) => {
          console.log(response.status);
        })
        .catch((err) => {
          console.log(err);
        });
    } else {
      alert('Email format wrong!');
    }
  }

  render() {
    return (
      <div className="container">
        <div className="login_links">
          <h2>Welcome to splitwise</h2>
          <span className="align-right">
            <div className="login_links">
              <a className="login" href="/login">
                Log in
              </a>

              <a className="signup" href="/signup">
                Sign up
              </a>
            </div>
          </span>
        </div>

        <div className="login-form">
          <div className="main-div">
            <div className="form-group">
              <label htmlFor="email">
                Email
                <input
                  type="email"
                  className="form-control"
                  name="email"
                  onChange={this.emailChangeHandler}
                  required
                />
              </label>
            </div>

            <div className="form-group">
              <label htmlFor="passowrd">
                Passowrd
                <input
                  type="password"
                  className="form-control"
                  name="password"
                  onChange={this.passwordChangeHandler}
                />
              </label>
            </div>
            <button type="button" onClick={this.submitLogin}>
              Log in
            </button>
          </div>
        </div>
      </div>
    );
  }
}

export default Login;
