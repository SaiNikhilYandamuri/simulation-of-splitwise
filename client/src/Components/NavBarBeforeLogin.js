import React from 'react';
import { Navbar, Button, Nav } from 'react-bootstrap';

function NavBarBeforeLogin() {
  return (
    <Navbar bg="success" expand="lg">
      <Navbar.Brand href="./landing">Splitwise</Navbar.Brand>
      <Nav className="mr-auto">
        <Nav.Link href="#home" />
      </Nav>
      <Button variant="outline-light" href="./login" data-testid="Login">
        Login
      </Button>
      <Button variant="warning" href="./signup">
        Signup
      </Button>
    </Navbar>
  );
}

export default NavBarBeforeLogin;
