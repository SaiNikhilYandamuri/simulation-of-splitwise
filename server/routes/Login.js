const router = require("express").Router();
const Users = require("../model/Users");
const jwt = require("jsonwebtoken");
const bcrypt = require("bcrypt");
const { auth } = require("../Utils/passport");
const secret = "hello";
auth();
// Validation

router.post("/login", async (req, res) => {
  console.log("inside login");
  try {
    const user = await Users.findOne({ email: req.body.email });
    if (!user) {
      return res.status(400).send("Enter Valid Credentials!");
    }
    const encryptedPassword = bcrypt.compare(req.body.password, user.password);
    if (!encryptedPassword) {
      return res.status(400).send("Enter Valid Credentials!");
    }
    const payload = {
      _id: user._id,
      username: user.fullname,
      email: user.email,
      currency: user.currency,
    };
    console.log(payload);
    const token = await jwt.sign(payload, secret, {
      expiresIn: 1000000,
    });
    console.log(token);
    res.status(200).send(token);
    //res.status(200).send("JWT " + token);
  } catch (err) {
    return res.status(400).json({ message: "Enter Valid credentials" });
  }
});
module.exports = router;
