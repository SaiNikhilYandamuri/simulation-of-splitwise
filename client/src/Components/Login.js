import axios from 'axios';
import { React, useState } from 'react';
import alert from 'alert';
import Button from 'react-bootstrap/Button';
import { useHistory } from 'react-router-dom';
import { useSelector, useDispatch } from 'react-redux';
import Form from 'react-bootstrap/Form';
import Jumbotron from 'react-bootstrap/esm/Jumbotron';
import logged from '../actions';
import NavBarBeforeLogin from './NavBarBeforeLogin';

function Login() {
  const [email, emailChangeHandler] = useState('');
  const [password, passwordChangeHandler] = useState('');
  const history = useHistory();

  const isLogged = useSelector((state) => state.isLogged);
  const dispatch = useDispatch();

  const loadSuccess = () => {
    history.push('/dashboard');
  };

  const submitLogin = (e) => {
    e.preventDefault();
    console.log('inside function');

    if (email.includes('@') && email.includes('.com')) {
      axios
        .post('http://localhost:4000/login', {
          email,
          password,
        })
        .then((response) => {
          console.log(response.status);
          console.log(isLogged);
          sessionStorage.setItem('email', response.data.email);
          sessionStorage.setItem('fullname', response.data.fullname);
          loadSuccess();
          dispatch(logged(response.data.fullname, response.data.email));
        })
        .catch((err) => {
          if (!err) alert(err.response.data.message);
        });
    } else {
      console.log('Email format wrong!');
    }
  };
  return (
    <div>
      <NavBarBeforeLogin />
      <Jumbotron>
        <center>
          <Form>
            <Form.Text className="text-muted">Welcome to Splitwise</Form.Text>

            <Form.Group controlId="formBasicEmail">
              <Form.Label>Email address</Form.Label>
              <Form.Control
                size="sm"
                type="email"
                placeholder="Enter email"
                onChange={(e) => {
                  emailChangeHandler(e.target.value);
                }}
              />
            </Form.Group>

            <Form.Group controlId="formBasicPassword">
              <Form.Label>Password</Form.Label>
              <Form.Control
                size="sm"
                type="password"
                placeholder="Password"
                onChange={(e) => {
                  passwordChangeHandler(e.target.value);
                }}
              />
            </Form.Group>

            <Button variant="warning" type="submit" onClick={submitLogin}>
              Log in
            </Button>
          </Form>
        </center>
      </Jumbotron>
    </div>
  );
}

export default Login;
