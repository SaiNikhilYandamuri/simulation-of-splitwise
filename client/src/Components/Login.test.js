/* eslint-disable react/prop-types */
/* eslint-disable no-undef */
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import TestRenderer from 'react-test-renderer';
import { createStore } from 'redux';
import { MemoryRouter } from 'react-router-dom';
// import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import '@testing-library/jest-dom';
import allReducers from '../reducers';
import Login from './Login';
import NavBarBeforeLogin from './NavBarBeforeLogin';

const store = createStore(allReducers);
const component = TestRenderer.create(
  <Provider store={store}>
    <MemoryRouter>
      <Login>
        <NavBarBeforeLogin />
      </Login>
    </MemoryRouter>
  </Provider>
);

// eslint-disable-next-line no-undef
afterEach(cleanup);

it('renders', async () => {
  expect(component.toJSON()).toMatchSnapshot();
});

test('Check for header', async () => {
  const { getByTestId } = render(
    <Provider store={store}>
      <MemoryRouter>
        <Login>
          <NavBarBeforeLogin />
        </Login>
      </MemoryRouter>
    </Provider>
  );
  expect(getByTestId('LoginButton')).toHaveTextContent('Log in');
});
