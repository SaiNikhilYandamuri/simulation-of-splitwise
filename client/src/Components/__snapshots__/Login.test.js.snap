// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div>
  <div>
    <nav
      className="navbar navbar-expand-lg navbar-light bg-success"
    >
      <a
        className="navbar-brand"
        href="./landing"
      >
        Splitwise
      </a>
      <div
        className="mr-auto navbar-nav"
        onKeyDown={[Function]}
      >
        <a
          className="nav-link"
          data-rb-event-key="#home"
          href="#home"
          onClick={[Function]}
          onKeyDown={[Function]}
        />
      </div>
      <a
        className="btn btn-outline-light"
        data-testid="Login"
        href="./login"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        Login
      </a>
      <a
        className="btn btn-warning"
        href="./signup"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        Signup
      </a>
    </nav>
    <div
      className="row"
    >
      <div
        className="col"
      />
      <div
        className="col"
      >
        <center>
          <form
            className=""
          >
            <small
              className="text-muted form-text"
            >
              <h4>
                Welcome to Splitwise
              </h4>
            </small>
            <div
              className="form-group"
            >
              <label
                className="form-label"
                htmlFor="formBasicEmail"
              >
                Email address
              </label>
              <input
                className="form-control form-control-sm"
                id="formBasicEmail"
                onChange={[Function]}
                placeholder="Enter email"
                type="email"
              />
            </div>
            <div
              className="form-group"
            >
              <label
                className="form-label"
                htmlFor="formBasicPassword"
              >
                Password
              </label>
              <input
                className="form-control form-control-sm"
                id="formBasicPassword"
                onChange={[Function]}
                placeholder="Password"
                type="password"
              />
            </div>
            <button
              className="btn btn-warning"
              data-testid="LoginButton"
              disabled={false}
              onClick={[Function]}
              type="submit"
            >
              Log in
            </button>
          </form>
        </center>
      </div>
      <div
        className="col"
      />
    </div>
  </div>
</div>
`;
